youtube_dl/postprocessor/ffmpeg.py

```diff
@@ -431,21 +431,36 @@ def run(self, information):
 
 
 class FFmpegMetadataPP(FFmpegPostProcessor):
+
+    def __init__(self, downloader=None, preferredinfo=None):
+        super(FFmpegMetadataPP, self).__init__(downloader)
+        self._preferredinfo = preferredinfo if isinstance(preferredinfo, dict) else None
+
     def run(self, info):
         metadata = {}
 
+        def add_info(meta_list, info_list, metadata, info_obj):
+            if info_obj is not None:
+                for info_f in info_list:
+                    if info_obj.get(info_f) is not None:
+                        for meta_f in meta_list:
+                            metadata[meta_f] = info_obj[info_f]
+                        return info_f
+            return None
+
         def add(meta_list, info_list=None):
             if not info_list:
                 info_list = meta_list
             if not isinstance(meta_list, (list, tuple)):
                 meta_list = (meta_list,)
             if not isinstance(info_list, (list, tuple)):
                 info_list = (info_list,)
-            for info_f in info_list:
-                if info.get(info_f) is not None:
-                    for meta_f in meta_list:
-                        metadata[meta_f] = info[info_f]
-                    break
+            preferred_key = add_info(meta_list, info_list, metadata, self._preferredinfo)
+            if preferred_key is None:
+                add_info(meta_list, info_list, metadata, info)
```

Comment:
`merge_dicts`.

Reply:
@dstftw My concern with a straight merge dict is if the info uses a different key from the info_list than the preferredinfo uses (e.g. info['track'] = 'a' and preferredinfo['title'] = 'b') - if they were merged without knowledge to the info_list, metadata['title'] would result in the track name from info instead of the title name from prefferredinfo. There are two options:

1. Leave as is
2. modify the preferredinfo object (on each add function call) to add the keys for all elements within the info_list, then preform a dictionary merge within each add onto a copied info

Let me know your preference and I will implement it.

Reply:
This is expected behavior, you named it preferredinfo not preferredmetadatatoembed. `track` from info must be preferred over `title` from preferredinfo.
Again: you merge `_preferredinfo` with `info` into separate merged info **not touching** the initial dicts then operate with old code on merged info instead of `info`.

Reply:
@dstftw I will rename the object to preferredmetadatatoembed and propagate throughout the code as this is the functionality that I believe should be implemented.

Reply:
No. In this case the whole new notion of embeddable meta fields should be introduced.

Reply:
This entire feature is about metadata override. The functionality is in place to do that. The name of the feature, the description of the feature, the class that it is being introduce into (FFmpegMetadataPP) are all about metadata. The only thing that is not in alignment with it is the name of the input parameter. What additional functionality and/or changes need to be put in place to allow for this to be the nature of the feature?

Reply:
It's about metadata however it does not reveal anything about its internals leaving it implementation detail and providing us more freedom. Once you start allowing to override concrete embeddable meta fields you must at least standardize and document the list of such fields in order for user to know what to actually override. That's actually not what I planned to maintain.

Reply:
@dstftw So the lists of metadata that are hard-coded within the Class right now are the same elements that this functionality allows to be overridden - no more and no less. My assumption was that this list was already being maintained?

Reply:
No. As already said it's an implementation detail and may be changed any time.

Reply:
@dstftw Ok - I understand. Is there any compromise where I can make the help method for the CLI argument dynamically expose the list of meta-data fields that are available for override?